std/algorithm/mutation.d

```diff
@@ -281,6 +307,14 @@ Elements can be swapped across ranges of different types:
         bringToFront(r1, r2) == 4 || assert(0);
         assert(equal(arr, [1, 2, 3, 4, 5, 6, 7]));
     }
+
+    // Bugzilla 16959
```

Comment:
Because it's specifically about char[], let's have a non-ASCII unittest here.

Related, this change is essentially the following, which leaves two illegal Unicode strings behind:
```
import std.algorithm;
import std.string;
import std.stdio;

void main() {
    auto ar = representation("a".dup);
    auto br = representation("ç".dup);

    bringToFront(ar, br);

    auto a = cast(char[])ar;
    auto b = cast(char[])br;

    assert(a == "\303");     // Illegal (incomplete) UTF-8
    assert(b == "\247a");    // Illegal UTF-8
}
```
Are we concerned about that in bringToFront or is that a case that the caller should be responsible for?